app/web/features/dashboard/Hero/useHeroBackgroundTheme.ts

```diff
@@ -0,0 +1,38 @@
+import { createTheme, Theme, useTheme } from "@material-ui/core";
+import { useMemo } from "react";
+
+const getTheme = (theme: Theme) =>
+  createTheme({
+    ...theme,
+    palette: {
+      ...theme.palette,
+      primary: {
+        main: theme.palette.common.white,
+      },
+      text: {
+        ...theme.palette.text,
+        primary: theme.palette.common.white,
+      },
+    },
+    shape: {
+      ...theme.shape,
+      borderRadius: 32, // px
+    },
+    typography: {
+      ...theme.typography,
+      button: {
+        ...theme.typography.button,
+        color: theme.palette.primary.main,
+        fontWeight: 700,
+      },
+    },
+  });
+
+/**
+ * Hook that returns a modified theme that adapts text to be over the hero image background
+ */
+export default function useHeroBackgroundTheme(): Theme {
+  const theme = useTheme();
+  const imageOverlayTheme = useMemo(() => getTheme(theme), [theme]);
```

Comment:
I haven't measured performance, but `createTheme` internally does a few deep merges and so I generally call it outside the component render (for the main theme) or memoizing the result. https://github.com/mui/material-ui/blob/v4.12.4/packages/material-ui/src/styles/createTheme.js

Reply:
I see. The `theme` is actually exported in `theme.ts` (at the root of the `web` project). Might it make more sense to import that directly and create the override theme directly from that outside of the component/React lifecycle?

Reply:
Yes, that's possible as well. It would be more performant, although it likely is too small to measure any difference. I prefer to leave it inside the React lifecycle so the `imageOverlayTheme` will get redone if `theme` got changed. I prefer to expect the theme to come from context than to be an import from this component that is somewhere inside the node tree, not wrapping everything at the top. For example, if we added later dark mode, we'd have to change a property on the `theme` being passed to the provider at the top of the tree. Let me know though if you prefer to make the theme outside, I can change it.

Reply:
I think it's important to do it from the context this way in case there is a `ThemeProvider` further down the tree

Reply:
Yeah good point about keeping it as a hook so React will correctly sync against `theme` changes! I did think about this but also since we don't support having multiple themes currently, we'd have to change how we pass the `theme` to the `ThemeProvider` later on anyway and I thought we could have just updated it to the hook then.

I suppose it probably doesn't make a big difference either way, so happy to leave it as is, as it's probably the more correct way to express that we want the theme to be reactive to changes.

Reply:
sounds good, leaving it as-is then 👍 thank you!